src/main/java/com/somemore/community/repository/board/CommunityBoardRepositoryImpl.java

```diff
@@ -38,18 +42,30 @@ public Optional<CommunityBoard> findById(Long id) {
     }
 
     @Override
-    public List<CommunityBoardView> getCommunityBoards() {
-        return getCommunityBoardsQuery()
+    public Page<CommunityBoardView> findCommunityBoards(Pageable pageable) {
+        List<CommunityBoardView> content = getCommunityBoardsQuery()
                 .where(QCommunityBoard.communityBoard.deleted.eq(false))
                 .fetch();
+
+        JPAQuery<Long> countQuery = queryFactory
+                .select(communityBoard.count())
+                .from(communityBoard);
+
+        return PageableExecutionUtils.getPage(content, pageable, countQuery::fetchOne);
```

Comment:
```suggestion
    public Page<CommunityBoardView> findCommunityBoards(Pageable pageable) {
        List<CommunityBoardView> content = getCommunityBoardsQuery()
                .where(QCommunityBoard.communityBoard.deleted.eq(false))
                .fetch();

        JPAQuery<Long> countQuery = queryFactory
                .select(communityBoard.count())
                .from(communityBoard);
                .where(deleted 조건)
                .fetchOne();

        return PageableExecutionUtils.getPage(content, pageable, countQuery);
```

삭제 조건과 fetchOne을 쿼리로 가져올 수 있을 것 같아요.
이후에도 삭제 조건이 자주 사용되던데 메서드로 추출해도 괜찮을 것 같습니다.

Reply:
>         LongSupplier countQuery = () -> Optional.ofNullable(queryFactory
>                         .select(communityBoard.count())
>                         .from(communityBoard)
>                         .where(communityBoard.deleted.eq(false))
>                         .fetchOne())
>                 .orElse(0L);
> 
>         return PageableExecutionUtils.getPage(content, pageable, countQuery);
> 

        
fecthOne을 쿼리로 가져오려면 null처리를 해줘야하네요
이렇게 수정하면 될까요?

삭제 메서드 추출했습니다!

Reply:
공급자를 사용해야하면 아영 님 원래 하신대로 `::fetchOne`을 사용하는 게 좋을 것 같습니다. 제가 사용하던 페이징이랑 다른거네요... 죄송합니다